File: src/bot/commands/pause.ts
import type { Message } from 'discord.js-selfbot-v13';
import { getVideoStreamer } from '../../stream/video-streamer.js';

export async function pauseCommand(message: Message, _args: string[]): Promise<void> {
  if (!message.guild) {
    await message.channel.send('❌ This command can only be used in a server');
    return;
  }

  const guildId = message.guild.id;
  const videoStreamer = getVideoStreamer();
  const session = videoStreamer.getSession(guildId);

  if (!session) {
    await message.channel.send('❌ Nothing is currently playing');
    return;
  }

  if (session.isPaused) {
    const success = await videoStreamer.resumeStream(guildId);
    if (success) {
      await message.channel.send('▶️ Playback resumed');
    } else {
      await message.channel.send('❌ Failed to resume playback');
    }
  } else {
    const success = await videoStreamer.pauseStream(guildId);
    if (success) {
      await message.channel.send('⏸️ Playback paused');
    } else {
      await message.channel.send('❌ Failed to pause playback');
    }
  }
}

File: selfbot/commands/pause.py
"""Pause/resume command via IPC."""

from __future__ import annotations

from typing import TYPE_CHECKING
import discord

from ..ipc import ipc

if TYPE_CHECKING:
    from ..client import SchroStreamBot


async def pause_command(bot: SchroStreamBot, message: discord.Message, args: list[str]) -> None:
    if not message.guild:
        await message.channel.send("❌ This command can only be used in a server")
        return

    result = await ipc.control_pause(str(message.guild.id))

    if result.get("success"):
        action = result.get("message", "Toggled")
        await message.channel.send(f"{'⏸️' if 'Pause' in action else '▶️'} {action}")
    else:
        await message.channel.send(f"❌ {result.get('error', 'Nothing is currently playing')}")
